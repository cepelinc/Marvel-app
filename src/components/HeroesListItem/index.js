import React from 'react';
import PropTypes from 'prop-types';
import style from './style.styled';
const Container = style.container;
const Link = style.link;

const buildAnswer = (data) => data ? 'yeah' : 'nope'

const HeroesListItem = ({name, comics, series, events, stories, picture, id, handleHero}) => (
  <Container>
    <div>
      <div>
        {picture ? <img src={`${picture.path}.${picture.extension}`}/> : 'no picture available'}
        <span>{name}</span>
      </div>
      <div>
        <span>Comics</span>
        <p>{buildAnswer(comics)}</p>
      </div>
      <div>
        <span>Series</span>
        <p>{buildAnswer(series)}</p>
      </div>
      <div>
        <span>Events</span>
        <p>{buildAnswer(events)}</p>
      </div>
      <div>
        <span>Stories</span>
        <p>{buildAnswer(stories)}</p>
      </div>
    </div>
    <Link onClick={() => handleHero(id)} to={`/heroes/${id}`}>Find out more about {name} &gt; &gt; &gt;</Link>
  </Container>
)

HeroesListItem.propTypes = {
  name: PropTypes.string,
  comics: PropTypes.number,
  series: PropTypes.number,
  events: PropTypes.number,
  stories: PropTypes.number,
  picture: PropTypes.object,
  id: PropTypes.number,
  handleHero: PropTypes.func

};


export default HeroesListItem
