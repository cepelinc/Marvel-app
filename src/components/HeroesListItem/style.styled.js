
import styled from 'styled-components'
import { Link } from "react-router-dom";

export default {
  container: styled.div`
    display: flex;
    flex-direction: column;
    margin: 5px;
    div{
      display: flex;
        div{
        align-items: center;
        justify-content: center;
        display: flex;
        flex-direction: column;
        padding: .2em;
        font-size: 1em;
        text-align: center;
        background: red;
        color: white;
        padding: 5px;
        border-left: 1px solid #E8E8E8;
        img {
          width: 100%
        };
        span {
          text-decoration: underline;
        };
        p {
          font-size: 1em;
        }
      };

    }
  `,
  link: styled(Link)`
    text-align: right;
    font-size: 1em;
    padding: 5px
  `,
}
