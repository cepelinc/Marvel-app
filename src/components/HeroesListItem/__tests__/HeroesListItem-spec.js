import React from 'react'
import Jest from 'jest';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import HeroesListItem from '@components/HeroesListItem'

describe('HeroesList', () => {
  const mockData = {
    name: 'Spiderman', id: 1, comics: 5, series: 5, events: 5, stories: 5, picture: {path: 'path', extension: 'jpg'}, offset: 1, handleHero : Jest.func,
  }
  it('matches snapshot', () => {
    const Component = renderer.create(
      <BrowserRouter>
        <HeroesListItem {...mockData} />
      </BrowserRouter>)
      .toJSON()

    expect(Component).toMatchSnapshot()
  })
})
