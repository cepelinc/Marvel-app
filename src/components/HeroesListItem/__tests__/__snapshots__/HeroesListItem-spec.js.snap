// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesList matches snapshot 1`] = `
<div
  className="stylestyled__container-u3r3w8-0 bsOeVg"
>
  <div>
    <div>
      <img
        src="path.jpg"
      />
      <span>
        Spiderman
      </span>
    </div>
    <div>
      <span>
        Comics
      </span>
      <p>
        yeah
      </p>
    </div>
    <div>
      <span>
        Series
      </span>
      <p>
        yeah
      </p>
    </div>
    <div>
      <span>
        Events
      </span>
      <p>
        yeah
      </p>
    </div>
    <div>
      <span>
        Stories
      </span>
      <p>
        yeah
      </p>
    </div>
  </div>
  <a
    className="stylestyled__link-u3r3w8-1 gMsAOd"
    href="/heroes/1"
    onClick={[Function]}
  >
    Find out more about 
    Spiderman
     &gt; &gt; &gt;
  </a>
</div>
`;
